test/lib/next-test-utils.js

```diff
@@ -170,6 +170,7 @@ export async function check (contentFn, regex) {
       const newContent = await contentFn()
       if (regex.test(newContent)) {
         found = true
+        clearTimeout(timeout)
```

Comment:
This makes sure we don't have to wait the remainder of the timeouts of passed tests when they finish.